File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-2.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        fp, sp = 0, 1

        while sp < len(nums):
            while sp < len(nums) and nums[fp] == nums[sp]:
                sp += 1

            if sp < len(nums):
                nums[fp + 1], nums[sp] = nums[sp], nums[fp + 1]
                fp += 1
            sp += 1

        return fp + 1

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.java
class Solution {
    public int subarraySum(int[] nums, int k) {
                Map<Integer, Integer> map = new HashMap<>();
        int result = 0;
        int currentSum = 0;
        map.put(0, 1);

        for (int i = 0; i < nums.length; i++) {
            currentSum += nums[i];
            if (map.containsKey(currentSum - k)) {
                result += map.get(currentSum - k);
            }
            map.put(currentSum, map.getOrDefault(currentSum, 0) + 1);
        }

        return result;
    }
}